src/GovUk.Education.ExploreEducationStatistics.Admin/Services/MigrateImportsService.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Models;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Cosmos.Table;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using static GovUk.Education.ExploreEducationStatistics.Common.Model.FileType;
+using static GovUk.Education.ExploreEducationStatistics.Content.Model.ImportStatus;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    /**
+     * Temporary service for migrating Import rows in table storage to the database
+     */
+    public class MigrateImportsService : IMigrateImportsService
+    {
+        private readonly ContentDbContext _contentDbContext;
+        private readonly ITableStorageService _tableStorageService;
+        private readonly IUserService _userService;
+        private readonly ILogger<MigrateImportsService> _logger;
+
+        public MigrateImportsService(ContentDbContext contentDbContext,
+            ITableStorageService tableStorageService,
+            IUserService userService,
+            ILogger<MigrateImportsService> logger)
+        {
+            _contentDbContext = contentDbContext;
+            _tableStorageService = tableStorageService;
+            _userService = userService;
+            _logger = logger;
+        }
+
+        public async Task<Either<ActionResult, Unit>> MigrateImports()
+        {
+            return await _userService.CheckCanRunMigrations()
+                .OnSuccessVoid(async () =>
+                {
+                    var tableImports = await GetTableImports();
+                    _logger.LogInformation($"Found {tableImports.Count} table imports");
+
+                    var imports = await tableImports.SelectAsync(async tableImport =>
+                        await GetImport(tableImport));
+
+                    _logger.LogInformation("Saving database imports");
+                    await _contentDbContext.AddRangeAsync(imports);
```

Comment:
I've just made a change so that we can rerun this without affecting any new import rows that are added by the Admin for newly uploaded queued imports.

There's a new `migrated` field. The migration won't run if any existing `migrated` rows already exist, but doesn't care about non-migrated rows.

If the migration doesn't run for any reason, we would fix up the problem, clear out all the migrated rows and re-run it, without affecting new uploads.

For the deploy, the importer functions will be disabled at a time that there's no imports taking place. We'll then check there's no Incomplete imports and clean them up if necessary before attempting the migration for the first time.

The migrated rows  will have `migrated=true` so we can differentiate between them and clean them out if anything goes wrong.

With this approach I think the Importer can be restarted after the deploy (double check there's no old-style messages on the queues first) even before the migration has been run.